### PR TITLE
added support for anaconda (conda) environment

### PR DESCRIPTION
**Pull request by Davidportlouis**
Added custom color & separate segment for anaconda (conda) environment
**image added for reference**
pr_img

### CODE REVIEW

File: trueline.sh [hunk 1]
```diff
@@ -405,6 +422,7 @@ unset TRUELINE_COLORS_DEFAULT
 
 if [[ "${#TRUELINE_SEGMENTS[@]}" -eq 0 ]]; then
     declare -a TRUELINE_SEGMENTS=(
+	'conda_env,black,purple,bold'
```
Comment (petobens) [inline trueline.sh:425]: Put this below venv, respect indentation and commented it out (i.e disable it by default).

**Comment by petobens** [comment]
Please add an entry to the readme and disable this segment by default.

File: trueline.sh (hunk 1 above)
Comment (Davidportlouis) [reply trueline.sh]: I added the above edits

File: README.md [hunk 2]
```diff
@@ -165,6 +165,7 @@ by default):
 | user           | enabled    | username and host (conditional on ssh status) |
 | venv           | enabled    | Python virtual environment                    |
 | working_dir    | enabled    | current working directory                     |
+| conda env	     | disabled   | Conda environment			                  |
```
Comment (petobens) [inline README.md:168]: Please align markers.

File: trueline.sh [hunk 3]
```diff
@@ -407,6 +425,7 @@ if [[ "${#TRUELINE_SEGMENTS[@]}" -eq 0 ]]; then
     declare -a TRUELINE_SEGMENTS=(
         'user,black,white,bold'
         'venv,black,purple,bold'
+	    # 'conda_env,black,soft_red,bold'
```
Comment (petobens) [inline trueline.sh:428]: Keep indentation exactly the same as the commented line below (the one about the newline segment)

File: README.md (hunk 2 above)
Comment (petobens) [reply README.md:168]: Allso move this un level up (i.e example below venv)

File: trueline.sh (hunk 3 above)
Comment (Davidportlouis) [reply trueline.sh:428]: I assume the indentation is same as the one for newline
rsz_screenshot_from_2020-08-02_06-29-48

Comment (Davidportlouis) [reply trueline.sh:428]: there seems to be some issues in GitHub PR view it's fine in my local machine as well as in my fork

File: README.md (hunk 2 above)
Comment (Davidportlouis) [reply README.md:167]: the unaligned marker is shown only in git diff, it renders aligned both in my local machine as well as raw code in my forked repo,
also I have moved the conda env below venv as specified in previous review.

**Comment by petobens** [comment]
Why is that our conda env variable check is different to https://github.com/chris-marsh/pureline/pull/31/files#diff-9c703c57e323b81e2bea06a75876ecafR266 ? (I don't use conda myself so no idea if one should be preferred over the other one).

**Comment by petobens** [comment]
> the unaligned marker is shown only in git diff, it renders aligned both in my local machine as well as raw code in my forked repo,

Is that so? So this raw file is wrong? https://raw.githubusercontent.com/petobens/trueline/bfe6ed77aac56637baaad24a21045ef5f16dd141/README.md

**Comment by Davidportlouis** [comment]
> 
> 
> > the unaligned marker is shown only in git diff, it renders aligned both in my local machine as well as raw code in my forked repo,
> 
> Is that so? So this raw file is wrong? https://raw.githubusercontent.com/petobens/trueline/bfe6ed77aac56637baaad24a21045ef5f16dd141/README.md

sorry didn't see that, now the  markers in README have been aligned

File: README.md [hunk 4]
```diff
@@ -164,8 +164,10 @@ by default):
 | read_only      | enabled    | indicator of read only directory              |
 | user           | enabled    | username and host (conditional on ssh status) |
 | venv           | enabled    | Python virtual environment                    |
+| conda_env      | disabled   | current anaconda environment                  |
 | working_dir    | enabled    | current working directory                     |
 
+
```
Comment (petobens) [inline README.md:170]: Please remove this extra line.

**Comment by petobens** [comment]
Just one last comment. I'll merge after that.

File: README.md (hunk 4 above)
Comment (Davidportlouis) [reply README.md:170]: ok

Comment (Davidportlouis) [reply README.md]: I removed the extra line in README

File: trueline.sh [hunk 5]
```diff
@@ -407,6 +425,7 @@ if [[ "${#TRUELINE_SEGMENTS[@]}" -eq 0 ]]; then
     declare -a TRUELINE_SEGMENTS=(
         'user,black,white,bold'
         'venv,black,purple,bold'
+        # 'conda_env,black,soft_red,bold'
```
Comment (petobens) [inline trueline.sh:428]: We don't have a `soft_red` color in the default colors. Change it to `purple` to be consistent with the standard python venv.

File: trueline.sh [hunk 6]
```diff
@@ -116,6 +116,24 @@ _trueline_venv_segment() {
     fi
 }
 
+_trueline_has_conda_env(){
```
Comment (petobens) [inline trueline.sh:119]: Add a space after the closing parenthesis

File: trueline.sh [hunk 7]
```diff
@@ -116,6 +116,24 @@ _trueline_venv_segment() {
...
+    printf "%s" "${CONDA_DEFAULT_ENV}"
+}
+
+_trueline_conda_env_segment() {
+    local conda_env=$(_trueline_has_conda_env)
```
Comment (petobens) [inline trueline.sh:124]: Wrap the right hand side in double quotes.

File: trueline.sh [hunk 8]
```diff
@@ -116,6 +116,24 @@ _trueline_venv_segment() {
     fi
 }
 
+_trueline_has_conda_env() {
+    printf "%s" "${CONDA_DEFAULT_ENV}"
+}
+
```
Comment (petobens) [inline trueline.sh:122]: Remove this line (the idea is to keep the same style we have with the venv)

File: trueline.sh [hunk 9]
```diff
@@ -116,6 +116,24 @@ _trueline_venv_segment() {
...
+        local fg_color="$1"
+        local bg_color="$2"
+        local font_style="$3"
+        local segment="$(_trueline_separator)"
+        segment+="$(_trueline_content "$fg_color" "$bg_color" "$font_style" " ${TRUELINE_SYMBOLS[venv]} $conda_env")"
+        PS1+="$segment"
+        _last_color=$bg_color
+    fi
+}
+
```
Comment (petobens) [inline trueline.sh:135]: Remove this line and we are ready to go.

File: trueline.sh (hunk 8 above)
Comment (Davidportlouis) [reply trueline.sh]: removed the line

File: trueline.sh (hunk 9 above)
Comment (Davidportlouis) [reply trueline.sh:134]: removed this line took, stylized exactly the same as venv

**Comment by petobens** [comment]
Thank you.

**Comment by Davidportlouis** [comment]
> 
> 
> Thank you.

Thanks for merging, looking forward to contributing more in future

**Pull request merged**